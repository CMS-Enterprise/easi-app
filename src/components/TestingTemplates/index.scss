.templates-for-508-testing {
  &__section {
    h2, p {
      margin-top: 0;
    }
  }
}
