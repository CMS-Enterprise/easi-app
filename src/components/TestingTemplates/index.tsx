import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  IconArrowForward,
  IconError,
  Link as UswdsLink,
  SummaryBox
} from '@trussworks/react-uswds';
import classNames from 'classnames';

import RemediationPlanDoc from 'assets/files/CMS508RemediationPlanTemplate.pdf';
import TestPlanDoc from 'assets/files/Section508TestPlanTemplate.pdf';
import HelpPageIntro from 'components/HelpPageIntro';
import UswdsReactLink from 'components/LinkWrapper';
import PageHeading from 'components/PageHeading';
import Divider from 'components/shared/Divider';
import { ArticleComponentProps } from 'types/articles';

import './index.scss';

export default ({ helpArticle, className }: ArticleComponentProps) => {
  const { t } = useTranslation(['accessibility', 'templatesFor508Testing']);

  const vpatConformanceLevels: { name: string; description: string }[] = t(
    'testingTemplates.vpatSection.subSection.item1.levels',
    {
      returnObjects: true
    }
  );
  const testPlanList: string[] = t(
    'testingTemplates.testPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  const remediationPlanList: string[] = t(
    'testingTemplates.remediationPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  return (
    <div className={classNames('templates-for-508-testing', className)}>
      {!helpArticle ? (
        <PageHeading className="line-height-heading-2">
          {t('testingTemplates.heading')}
        </PageHeading>
      ) : (
        <HelpPageIntro
          type="Section 508"
          heading={t('testingTemplates.heading')}
          subheading={t('templatesFor508Testing:description')}
        />
      )}
      <div className="margin-bottom-6 padding-2 bg-base-lightest">
        <h3>{t('templatesFor508Testing:pageContents')}</h3>
        <ul className="usa-list usa-list--unstyled">
          <li className="margin-bottom-1">
            <UswdsLink href="#vpat">
              {t('testingTemplates.vpatSection.heading')}
              <IconArrowForward />
            </UswdsLink>
          </li>
          <li className="margin-bottom-1">
            <UswdsLink href="#test-plan">
              {t('testingTemplates.testPlanSection.heading')}
              <IconArrowForward />
            </UswdsLink>
          </li>
          <li>
            <UswdsLink href="#remediation-plan">
              {t('testingTemplates.remediationPlanSection.heading')}
              <IconArrowForward />
            </UswdsLink>
          </li>
        </ul>
      </div>
      <div className="templates-for-508-testing__section">
        <h2 id="vpat" className="margin-bottom-2">
          {t('testingTemplates.vpatSection.heading')}
        </h2>
        <p className="margin-bottom-4">
          {t('testingTemplates.vpatSection.description')}
        </p>
        <h3 className="margin-bottom-2">
          {t('testingTemplates.vpatSection.subSection.heading')}
        </h3>
        <p className="margin-bottom-1">
          {t('testingTemplates.vpatSection.subSection.item1.text')}
        </p>
        <ul className="usa-list usa-list--unstyled desktop:grid-col-8">
          {vpatConformanceLevels.map(level => (
            <li key={level.name} className="maxw-none margin-bottom-1">
              <span className="text-bold display-inline">{level.name}</span>{' '}
              {level.description}
            </li>
          ))}
        </ul>
        <p className="margin-bottom-4">
          {t('testingTemplates.vpatSection.subSection.item2.text')}
        </p>
        <SummaryBox
          heading={t(
            'testingTemplates.vpatSection.subSection.downloadVPAT.heading'
          )}
          className="desktop:grid-col-6 margin-bottom-6"
        >
          <p>
            <UswdsReactLink
              to="https://www.itic.org/policy/accessibility/vpat"
              target="_blank"
              rel="noopener noreferrer"
              variant="external"
            >
              {t(
                'testingTemplates.vpatSection.subSection.downloadVPAT.line1.linkText'
              )}
            </UswdsReactLink>
            {` `}
            {t(
              'testingTemplates.vpatSection.subSection.downloadVPAT.line1.otherText'
            )}
          </p>
          <p className="display-flex flex-row flex-align-center">
            <IconError className="margin-right-1" />
            {` `}
            {t(
              'testingTemplates.vpatSection.subSection.downloadVPAT.line2.text'
            )}
          </p>
          <p>
            <UswdsReactLink
              to="https://www.youtube.com/watch?v=kAkSV9xiJ1A"
              target="_blank"
              rel="noopener noreferrer"
              variant="external"
            >
              {t(
                'testingTemplates.vpatSection.subSection.downloadVPAT.line3.linkText'
              )}
            </UswdsReactLink>
          </p>
        </SummaryBox>
        <Divider className="margin-bottom-4" />
      </div>
      <div className="templates-for-508-testing__section">
        <h2 id="test-plan" className="margin-bottom-2">
          {t('testingTemplates.testPlanSection.heading')}
        </h2>
        <p className="margin-bottom-2 line-height-sans-5 desktop:grid-col-9">
          {t('testingTemplates.testPlanSection.description')}
        </p>
        <ul className="usa-list usa-list--unstyled desktop:grid-col-8">
          {testPlanList.map(item => (
            <li key={item} className="maxw-none margin-bottom-1">
              {item}
            </li>
          ))}
        </ul>
        <SummaryBox
          heading={t('testingTemplates.testPlanSection.download.heading')}
          className="desktop:grid-col-6 margin-bottom-6"
        >
          <UswdsReactLink to={TestPlanDoc} target="_blank" variant="external">
            {t('testingTemplates.testPlanSection.download.link')}
          </UswdsReactLink>
        </SummaryBox>
        <Divider className="margin-bottom-4" />
      </div>
      <div className="templates-for-508-testing__section">
        <h2 id="remediation-plan" className="margin-bottom-2">
          {t('testingTemplates.remediationPlanSection.heading')}
        </h2>
        <p className="margin-bottom-2 line-height-sans-5 desktop:grid-col-9">
          {t('testingTemplates.remediationPlanSection.description')}
        </p>
        <ul className="usa-list usa-list--unstyled desktop:grid-col-8">
          {remediationPlanList.map(item => (
            <li key={item} className="maxw-none margin-bottom-1">
              {item}
            </li>
          ))}
        </ul>
        <SummaryBox
          heading={t(
            'testingTemplates.remediationPlanSection.download.heading'
          )}
          className="desktop:grid-col-6 margin-bottom-6"
        >
          <UswdsReactLink
            to={RemediationPlanDoc}
            target="_blank"
            variant="external"
          >
            {t('testingTemplates.testPlanSection.download.link')}
          </UswdsReactLink>
        </SummaryBox>
      </div>
    </div>
  );
};
