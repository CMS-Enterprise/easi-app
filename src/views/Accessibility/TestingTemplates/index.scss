

.accessibility-testing-templates {
  &__table-of-contents {
    padding-left: 0;
    list-style-type: none;
    li {
      &:before {
          content: "- ";
      }
    }
  }
  &__vpat-list, &__test-plan-list, &__remediation-plan-list {
    padding-left: 20px;
  }
  &__downloadBox {
    background: color($theme-color-info-lighter);
    padding: 24px 18px;
  }
  &__side-col {
  border-top: 4px solid $easi-bright-blue;
  }
}
