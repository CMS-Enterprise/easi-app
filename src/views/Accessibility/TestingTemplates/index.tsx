import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from '@trussworks/react-uswds';

import BreadcrumbNav from 'components/BreadcrumbNav';
import PageHeading from 'components/PageHeading';

import './index.scss';

const TestingTemplates = () => {
  const { t } = useTranslation('accessibility');
  const vpatConformanceLevels: { name: string; description: string }[] = t(
    'testingTemplates.vpatSection.subSection.item1.levels',
    {
      returnObjects: true
    }
  );
  const testPlanList: string[] = t(
    'testingTemplates.testPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  const remediationPlanList: string[] = t(
    'testingTemplates.remediationPlanSection.itemsToProvide',
    {
      returnObjects: true
    }
  );

  const tableOfContents = (
    <div className="accessibility-testing-templates">
      <p className="margin-bottom-1">Page contents</p>
      <ul className="accessibility-testing-templates__table-of-contents">
        <li>
          <Link href="#vpat">{t('testingTemplates.vpatSection.heading')}</Link>
        </li>
        <li>
          <Link href="#test-plan">
            {t('testingTemplates.testPlanSection.heading')}
          </Link>
        </li>
        <li>
          <Link href="#remediation-plan">
            {t('testingTemplates.remediationPlanSection.heading')}
          </Link>
        </li>
      </ul>
    </div>
  );

  const testPlanSection = (
    <div>
      <h2 id="test-plan">{t('testingTemplates.testPlanSection.heading')}</h2>
      <p>{t('testingTemplates.testPlanSection.description')}</p>
      <ul className="accessibility-testing-templates__test-plan-list">
        {testPlanList.map(item => (
          <li key={item}>{item}</li>
        ))}
      </ul>
    </div>
  );
  const remediationPlanSection = (
    <div>
      <h2 id="remediation-plan">
        {t('testingTemplates.remediationPlanSection.heading')}
      </h2>
      <p>{t('testingTemplates.remediationPlanSection.description')}</p>
      <ul className="accessibility-testing-templates__remediation-plan-list">
        {remediationPlanList.map(item => (
          <li key={item}>{item}</li>
        ))}
      </ul>
    </div>
  );

  const downloadVPAT = (
    <div className="accessibility-testing-templates__downloadBox">
      <h3 className="margin-top-0">
        {t('testingTemplates.vpatSection.subSection.downloadVPAT.heading')}
      </h3>
      <p>
        <Link
          href="https://www.itic.org/policy/accessibility/vpat"
          target="_blank"
          rel="noopener noreferrer"
        >
          {t(
            'testingTemplates.vpatSection.subSection.downloadVPAT.line1.linkText'
          )}
        </Link>
        {` `}
        {t(
          'testingTemplates.vpatSection.subSection.downloadVPAT.line1.otherText'
        )}
      </p>
      <p className="display-flex flex-row flex-align-center accessibility-testing-templates__alert-note">
        <i className="fa fa-exclamation-circle margin-right-1" />
        {` `}
        {t('testingTemplates.vpatSection.subSection.downloadVPAT.line2.text')}
      </p>
      <p>
        <Link
          href="https://www.youtube.com/watch?v=kAkSV9xiJ1A"
          target="_blank"
          rel="noopener noreferrer"
        >
          {t(
            'testingTemplates.vpatSection.subSection.downloadVPAT.line3.linkText'
          )}
        </Link>
        {` `}
        {t(
          'testingTemplates.vpatSection.subSection.downloadVPAT.line3.otherText'
        )}
      </p>
    </div>
  );

  const vpatSection = (
    <div>
      <h2 id="vpat">{t('testingTemplates.vpatSection.heading')}</h2>
      <p>{t('testingTemplates.vpatSection.description')}</p>
      <h3>{t('testingTemplates.vpatSection.subSection.heading')}</h3>
      <ul className="accessibility-testing-templates__vpat-list">
        <li>{t('testingTemplates.vpatSection.subSection.item1.text')}</li>
        <div className="padding-left-2">
          <dl>
            {vpatConformanceLevels.map(level => (
              <div key={level.name}>
                <dt className="text-bold display-inline">{level.name}</dt>{' '}
                <dd className="margin-left-0 display-inline">
                  {level.description}
                </dd>
              </div>
            ))}
          </dl>
        </div>
        <li>{t('testingTemplates.vpatSection.subSection.item2.text')}</li>
      </ul>
      {downloadVPAT}
    </div>
  );

  return (
    <div className="grid-container accessibility-testing-templates">
      <BreadcrumbNav className="margin-y-2">
        <li>
          <Link href="/">Home</Link>
          <i className="fa fa-angle-right margin-x-05" aria-hidden />
        </li>
        <li>Templates for 508 testing</li>
      </BreadcrumbNav>
      <div className="grid-row grid-gap-lg margin-top-6">
        <div className="grid-col-9 line-height-body-4">
          <div className="tablet:grid-col-10">
            <PageHeading className="margin-top-0">
              {t('testingTemplates.heading')}
            </PageHeading>
            {tableOfContents}
            {vpatSection}
            {testPlanSection}
            {remediationPlanSection}
          </div>
        </div>
        <div className="grid-col-3 accessibility-testing-templates__sidebar">
          <div>
            <h4>Need help? Contact the Section 508 team</h4>
            <Link href="mailto:CMS_Section508@cms.hhs.gov">
              CMS_Section508@cms.hhs.gov
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default TestingTemplates;
