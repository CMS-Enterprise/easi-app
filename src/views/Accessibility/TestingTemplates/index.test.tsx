import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import TestingTemplates from './index';

describe('TestingTemplates', () => {
  it('renders without crashing', () => {
    const component = render(<TestingTemplates />);
    expect(component.length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<TestingTemplates />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
