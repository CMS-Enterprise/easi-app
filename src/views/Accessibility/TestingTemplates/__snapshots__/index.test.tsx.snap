// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestingTemplates matches the snapshot 1`] = `
<div
  className="grid-container accessibility-testing-templates"
>
  <nav
    aria-label="breadcrumbs"
    className="easi-breadcrumb-nav margin-y-2"
    role="navigation"
  >
    <ol>
      <li>
        <a
          className="usa-link"
          href="/"
        >
          Home
        </a>
        <i
          aria-hidden={true}
          className="fa fa-angle-right margin-x-05"
        />
      </li>
      <li>
        Templates for 508 testing
      </li>
    </ol>
  </nav>
  <div
    className="grid-row grid-gap-lg margin-top-6"
  >
    <div
      className="grid-col-9 line-height-body-4"
    >
      <div
        className="tablet:grid-col-10"
      >
        <h1
          aria-live="polite"
          className="easi-h1 margin-top-0"
          tabIndex={-1}
        >
          Templates for 508 testing
        </h1>
        <div
          className="accessibility-testing-templates"
        >
          <p
            className="margin-bottom-1"
          >
            Page contents
          </p>
          <ul
            className="accessibility-testing-templates__table-of-contents"
          >
            <li>
              <a
                className="usa-link"
                href="#vpat"
              >
                Voluntary Product Assessment Template (VPAT)
              </a>
            </li>
            <li>
              <a
                className="usa-link"
                href="#test-plan"
              >
                Test Plan
              </a>
            </li>
            <li>
              <a
                className="usa-link"
                href="#remediation-plan"
              >
                Remediation Plan
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h2
            id="vpat"
          >
            Voluntary Product Assessment Template (VPAT)
          </h2>
          <p>
            A VPAT s a document that helps CMS and other government agencies understand how you will meet the 508 Standards for IT accessibility.
          </p>
          <h3>
            Tips to complete a VPAT
          </h3>
          <ul
            className="accessibility-testing-templates__vpat-list"
          >
            <li>
              For each row in the table, indicate the conformance level. There are 5 levels:
              <div
                className="padding-left-2"
              >
                <dl
                  title="Conformance levels"
                >
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Supports:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The functionality of the product has at least one method that meets the criterion without known defects or meets with equivalent facilitation.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Partially Supports:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      Some functionality of the product does not meet the criterion.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Does Not Support:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The majority of product functionality does not meet the criterion.
                    </dd>
                  </div>
                  <div>
                    <dt
                      className="text-bold display-inline"
                    >
                      Not Evaluated:
                    </dt>
                     
                    <dd
                      className="margin-left-0 display-inline"
                    >
                      The product has not been evaluated against the criterion. This can be used only in WCAG 2.0 Level AAA.
                    </dd>
                  </div>
                </dl>
              </div>
            </li>
            <li>
              After indicating the conformance level, provide a detailed explanation in the ‘Remarks and Evaluation’ column.
            </li>
          </ul>
          <div
            className="accessibility-testing-templates__downloadBox"
          >
            <h3
              className="margin-top-0"
            >
              Download VPAT template
            </h3>
            <p>
              <a
                className="usa-link"
                href="https://www.itic.org/policy/accessibility/vpat"
                rel="noopener noreferrer"
                target="_blank"
              >
                Download the current VPAT template (opens link in new tab)
              </a>
               
              from the Information Technology Industry Council (ITI) website.
            </p>
            <p
              className="display-flex flex-row flex-align-center accessibility-testing-templates__alert-note"
            >
              <i
                className="fa fa-exclamation-circle margin-right-1"
              />
               
              Do not skip any row in the document. This will result in delays.
            </p>
            <p>
              <a
                className="usa-link"
                href="https://www.youtube.com/watch?v=kAkSV9xiJ1A"
                rel="noopener noreferrer"
                target="_blank"
              >
                Watch the tutorial (opens link in a new tab)
              </a>
               
              on how to fill out a VPAT.
            </p>
          </div>
        </div>
        <div>
          <h2
            id="test-plan"
          >
            Test Plan
          </h2>
          <p>
            The Test Plan is a document that helps the Section 508 team understand your system and know what areas of your system to test before it goes live. In this document you will provide:
          </p>
          <ul
            className="accessibility-testing-templates__test-plan-list"
          >
            <li>
              a description of your system
            </li>
            <li>
              information on how to set up and access it
            </li>
            <li>
              user scenarios and steps taken to perform actions
            </li>
            <li>
              and any other constraints, assumptions or risks
            </li>
          </ul>
        </div>
        <div>
          <h2
            id="remediation-plan"
          >
            Remediation Plan
          </h2>
          <p>
            The Remediation Plan is a document that helps the Section 508 team understand how you plan to fix the issues that were identified from the 508 test. In this document you will provide:
          </p>
          <ul
            className="accessibility-testing-templates__remediation-plan-list"
          >
            <li>
              solutions for each issue
            </li>
            <li>
              when and how you plan to schedule these changes into your project
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="grid-col-3 accessibility-testing-templates__sidebar"
    >
      <div>
        <h4>
          Need help? Contact the Section 508 team
        </h4>
        <a
          className="usa-link"
          href="mailto:CMS_Section508@cms.hhs.gov"
        >
          CMS_Section508@cms.hhs.gov
        </a>
      </div>
    </div>
  </div>
</div>
`;
